src/vault.ts

```diff
@@ -0,0 +1,235 @@
+import { Address } from "@graphprotocol/graph-ts";
+import {
+  Deposited,
+  StrategySet,
+  Withdrawn,
+} from "../generated/VaultFactory/Vault";
+import { BIG_INT_ONE } from "./constants";
+import { loadBundle, loadToken, loadUser, loadVaultDayData } from "./entities";
+import {
+  createVaultDeposit,
+  createVaultWithdraw,
+  loadVault,
+  loadVaultUserPosition,
+} from "./entities/vault";
+import { loadVaultFactory } from "./entities/vaultFactory";
+import { loadVaultStrategy } from "./entities/vaultStrategy";
+import { formatTokenAmountByDecimals } from "./utils";
+import { updateAvaxInUsdPricing } from "./utils/pricing";
+import { Vault as VaultABI } from "../generated/VaultFactory/Vault";
+
+export function handleDeposited(event: Deposited): void {
+  const vault = loadVault(event.address);
+  if (!vault) {
+    return;
+  }
+
+  const vaultContract = VaultABI.bind(event.address);
+
+  // reset tvl aggregates until new amounts calculated
+  const factory = loadVaultFactory(Address.fromString(vault.factory));
+  factory.totalValueLockedAVAX = factory.totalValueLockedAVAX.minus(
+    vault.totalValueLockedAVAX
+  );
+
+  // load price bundle
+  updateAvaxInUsdPricing();
+  const bundle = loadBundle();
+
+  // load user
+  loadUser(event.params.user);
+
+  // get tokens
+  const tokenX = loadToken(Address.fromString(vault.tokenX));
+  const tokenY = loadToken(Address.fromString(vault.tokenY));
+
+  // get deposited amounts
+  const amountX = formatTokenAmountByDecimals(
+    event.params.amountX,
+    tokenX.decimals
+  );
+  const amountY = formatTokenAmountByDecimals(
+    event.params.amountY,
+    tokenY.decimals
+  );
+  const amountUSD = amountX
+    .times(tokenX.derivedAVAX)
+    .plus(amountY.times(tokenY.derivedAVAX))
+    .times(bundle.avaxPriceUSD);
+
+  // update vault total balance
+  const vaultBalances = vaultContract.try_getBalances().value;
+  vault.totalBalanceX = formatTokenAmountByDecimals(
+    vaultBalances.getAmountX(),
+    tokenX.decimals
+  );
+  vault.totalBalanceY = formatTokenAmountByDecimals(
+    vaultBalances.getAmountY(),
+    tokenY.decimals
+  );
```

Comment:
perhaps we should increment the vault balances using `amountX` and `amountY`, rather than fetching these from the contract every time.

Reply:
the thing is the strategy can adjust the LB position anytime and the amounts deposited will be different than the actual amounts of the current position

Reply:
I'm asking Louis

Reply:
Yeah so Louis confirmed we can't just increment the balances, it won't be the real balances held by the vault. If we are concerned by indexing speed, we can choose not to store it